src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -2789,6 +2789,12 @@ async def do_delete(self, id):
                     f'This certificate authority is in use by {text} service and cannot be deleted.'
                 )
 
+        if certs := (await self.middleware.call('certificate.query', [['issuer.id', '=', id]])):
```

Comment:
We can also have intermediate CAs, so let's please add that check as well. Thanks!